eip-5564.md

```diff
@@ -145,15 +145,79 @@ contract IERC5564Messenger {
+The `ERC5564Messenger` contract inherits the  `StakeManager` contract allowing users to stake ETH that is uses as an anti-DoS measure. More details in the ... section.
+```solidity
+/// @notice Interface for the Stake Manager contract.
+contract StakeManager{
+  
+  /// @dev Emitted when stake is deposited.
+  /// @param account The address of the staker who deposited the stake.
+  /// @param totalStaked The new total amount of staked tokens for the account.
+  event StakeDeposit (
+        address indexed account,
+        uint256 totalStaked
+  );
+    
+  /// @dev Emitted when stake is withdrawn.
+  /// @param account The address of the staker who withdrew the stake.
+  /// @param withdrawAddress The address to which the withdrawn amount was sent.
+  /// @param amount The amount of tokens withdrawn.
+  event StakeWithdrawal (
+        address indexed account,
+        address withdrawAddress,
+        uint256 amount
+  );
+  
+  /**
+   * @notice Returns the stake of the account.
+   * @param account The address of the staker to check the stake for.
+   * @return uint256 The amount of staked tokens for the account.
+   */
+  function balanceOf(address account) external view returns (uint256);
+
+  /**
+   * @notice Adds the specified amount to the account's stake.
+   * @dev The function is payable, so the amount is sent as value with the transaction.
+   * @param staker The address of the staker to add the stake for.
+   */
+  function addStake(address staker) external payable;
+  
+  /**
+   * @notice Withdraws the stake for the caller and sends it to the specified address.
+   * @param withdrawAddress The address to send the withdrawn amount to.
+   */
+  function withdrawStake(address payable withdrawAddress) external;
+}
+```
+
+
+### Stealth meta-address format
+
+The new address format for the stealth meta-addresses is based on [ERC-3770](./eip-3770.md) and extends it by adding a `st:` (*stealth*) as prefix.
+Thus, stealth meta-addresses on Ethereum come with the following format:
+
+```
+st:eth:0x<spendingKey><viewingKey>
+``` 
+
+*Notably, the address-format is only used to differentiate stealth addresses from standard addresses, however, the prefix is sliced away before doing any computations on the stealth meta-address.*
+
+---
 
-An example stealth address scheme for elliptic curves is below. Other stealth schemes may need to modify this approach. 
+### Initial Implementation of SECP256k1 with View Tags
+
+This ERC provides a foundation that is not tied to any specific cryptographic system through the IERC5564Messenger contract. In addition, it introduces the first implementation of a stealth address scheme that utilizes the SECP256k1 elliptic curve and`view tags`. The SECP256k1 elliptic curve is defined with the equation $y^2 = x^3 + 7 \pmod{p}$, where $p = 2^{256} - 2^{32} - 977$.
+
+The following reference is divided into a stealth address generation, parsing and stealth private key derivation section and is defined as follows:
```

Comment:
```suggestion
The following reference is divided into three sections:
1. Stealth address generation
2. Parsing announcements
3. Stealth private key derivation

 Definitions:
```

Reply:
We should also link to that page that gives secp256k1 the initial scheme ID in this section

Reply:
Oh yeah, I'll include it